Update Access API documentation section (#877)
onflow/docs: docs/networks/node-ops/access-onchain-data/access-nodes/accessing-data/access-api.md

---

### GetFullCollectionByID

> `GetFullCollectionByID` gets a batch of [transactions](#transaction) by ID.

Reviewer: ```suggestion
`GetFullCollectionByID` gets a collection by ID, which contains a set of [transactions](#transaction).
```

```proto
rpc GetFullCollectionByID(GetFullCollectionByIDRequest) returns (FullCollectionResponse);
```

#### Request

```proto
message GetFullCollectionByIDRequest {

>   bytes id

Reviewer: ```suggestion
  bytes id = 1;
```

}
```

### GetTransactionsByBlockID

`GetTransactionsByBlockID` gets all the [transactions](#transaction) for a specified block.

> 
> If the transactions are not found in the access node cache, the request is forwarded to a collection node.
> 
> _Currently, only transactions within the current epoch can be queried._
> 

Reviewer: neither of these are true. where did they come from?
```suggestion
```

```proto
rpc GetTransactionsByBlockID(GetTransactionsByBlockIDRequest) returns (TransactionsResponse);
```

#### Request

```proto
message GetTransactionsByBlockIDRequest {
  bytes block_id = 1;
}
```

#### Response

```proto
message TransactionsResponse {

>   repeated entities.Transaction transactions = 1;

Reviewer: ```suggestion
  repeated entities.Transaction transactions = 1;
  entities.Metadata metadata = 2;
```

  repeated flow.Event events
}
```
### GetTransactionResultByIndex

`GetTransactionResultByIndex` gets the execution result of a transaction at a specified block and index.

```proto
rpc GetTransactionResultByIndex(GetTransactionByIndexRequest) returns (TransactionResultResponse);
```

#### Request

```proto
message GetTransactionByIndexRequest {
  bytes block_id = 1;

>   uint32 index = 2;

Reviewer: This field is likely missing from other endpoints as well. Can you compare the docs to the protobuf to see if there are any other args missing?
```suggestion
  uint32 index = 2;
  entities.EventEncodingVersion event_encoding_version = 3;
```

}
```

#### Response

```proto
message TransactionResultResponse {
  flow.TransactionStatus status
  uint32 status_code
  string error_message
  repeated flow.Event events
}
```

### GetTransactionResultsByBlockID

`GetTransactionResultsByBlockID` gets all the transaction results for a specified block.

```proto
rpc GetTransactionResultsByBlockID(GetTransactionsByBlockIDRequest) returns (TransactionResultsResponse);
```

#### Request

```proto
message GetTransactionsByBlockIDRequest {
  bytes block_id = 1;
  entities.EventEncodingVersion event_encoding_version = 2;
}
```

#### Response

```proto
message TransactionResultResponse {
 repeated TransactionResultResponse transaction_results = 1;
}
```

### GetSystemTransaction

> `GetSystemTransaction` gets a system transaction.

Reviewer: ```suggestion
`GetSystemTransaction` gets the system transaction for a block.
```

```proto
rpc GetSystemTransaction(GetSystemTransactionRequest) returns (TransactionResponse);
```

#### Request

```proto
message GetSystemTransactionRequest {
  bytes block_id = 1;
}
```

#### Response

```proto
message TransactionResponse {
 entities.Transaction transaction = 1;
}
```

### GetSystemTransactionResult

> `GetSystemTransactionResult` gets a system transaction result for a specified block.

Reviewer: ```suggestion
`GetSystemTransactionResult` gets the system transaction result for a block.
```

}
```

### GetAccountBalanceAtLatestBlock

> `GetAccountAtBlockHeight` gets an account balance by address from the latest sealed execution state.

Reviewer: ```suggestion
`GetAccountBalanceAtLatestBlock` gets an account's balance by address from the latest sealed block.
```

> 
> The access node queries an execution node for the account details, which are stored as part of the execution state.

Reviewer: this level of detail isn't needed in the docs
```suggestion
```

```proto
rpc GetAccountBalanceAtLatestBlock(GetAccountBalanceAtLatestBlockRequest) returns (AccountBalanceResponse);
```

#### Request

```proto
message GetAccountBalanceAtLatestBlockRequest {
  bytes address = 1
}
```

#### Response

```proto
message AccountBalanceResponse {
  uint64 balance = 1;
}
```

### GetAccountBalanceAtBlockHeight

> `GetAccountAtBlockHeight` gets an account balance by address at the given block height.

Reviewer: ```suggestion
`GetAccountBalanceAtBlockHeight` gets an account's balance by address at the given block height.
```

```proto
rpc GetAccountBalanceAtBlockHeight(GetAccountBalanceAtBlockHeightRequest) returns (AccountBalanceResponse);
```

#### Request

```proto
message GetAccountBalanceAtBlockHeightRequest {
  bytes address = 1;
  uint64 block_height = 2;
}
```

#### Response

```proto
message AccountBalanceResponse {
  uint64 balance = 1;
}
```

### GetAccountKeyAtLatestBlock

> `GetAccountKeyAtLatestBlock` gets an account public key by address and key index from the latest sealed execution state.

Reviewer: ```suggestion
`GetAccountKeyAtLatestBlock` gets an account's public key by address and key index from the latest sealed block.
```

```proto
rpc GetAccountKeyAtLatestBlock(GetAccountKeyAtLatestBlockRequest) returns (AccountKeyResponse);
```

#### Request

```proto
message GetAccountKeyAtLatestBlockRequest {
  bytes address = 1;

>   // index of key to return

Reviewer: I'd say either include comments for all arguments or remove them from all.

  uint32 index = 2;
}
```

#### Response

```proto
message AccountKeyResponse {
  entities.AccountKey account_key = 1;
}
```

### GetAccountKeyAtBlockHeight

> `GetAccountKeyAtBlockHeight` gets an account public key by address and key index at the given block height.

Reviewer: ```suggestion
`GetAccountKeyAtBlockHeight` gets an account's public key by address and key index at the given block height.
```

```proto
rpc GetAccountKeyAtBlockHeight(GetAccountKeyAtBlockHeightRequest) returns (AccountKeyResponse);
```

#### Request

```proto
message GetAccountKeyAtBlockHeightRequest {
  bytes address = 1;
  uint64 block_height = 2;
  // index of key to return
  uint32 index = 3;
}
```

#### Response

```proto
message AccountKeyResponse {
  entities.AccountKey account_key = 1;
}
```

### GetAccountKeysAtLatestBlock

> `GetAccountKeysAtLatestBlock` gets an account public keys by address from the latest sealed execution state.

Reviewer: ```suggestion
`GetAccountKeysAtLatestBlock` gets an account's public keys by address from the latest sealed block.
```

```proto
rpc GetAccountKeysAtLatestBlock(GetAccountKeysAtLatestBlockRequest) returns (AccountKeysResponse);
```

#### Request

```proto
message GetAccountKeysAtLatestBlockRequest {
  bytes address = 1;
  // index of key to return
  uint32 index = 2;
}
```

#### Response

```proto
message AccountKeysResponse {
  repeated entities.AccountKey account_keys = 1;
}
```

### GetAccountKeysAtBlockHeight

> `GetAccountKeysAtBlockHeight` gets an account public keys by address at the given block height.

Reviewer: ```suggestion
`GetAccountKeysAtBlockHeight` gets an account's public keys by address at the given block height.
```

---

### GetNodeVersionInfo

> `GetNodeVersionInfo` node version information, such as semver, commit, sporkID and protocol version.

Reviewer: ```suggestion
`GetNodeVersionInfo` gets information about a node's current versions.
```

  bytes proposer_id = 9;
  bytes proposer_sig_data = 10;
  string chain_id = 11;
  bytes parent_voter_indices = 12;
  TimeoutCertificate last_view_tc = 13;
  uint64 parent_view = 14;
}
```

| Field                 | Description                                                                                                       |
|-----------------------|-------------------------------------------------------------------------------------------------------------------|
| id                    | SHA3-256 hash of the entire block payload                                                                         |
| parent_id             | ID of the previous block in the chain                                                                             |
| height                | Height of the block in the chain                                                                                  |
| timestamp             | The time at which this block was proposed                                                                         |
| payload_hash          | A hash of the payload of this block                                                                               |

> | view                  | Number at which this block was proposed.                                                                          |

Reviewer: ```suggestion
| view                  | View number during which this block was proposed.                                                                          |
```

| chunks             | Zero or more chunks                                  |
| service_events     | Zero or more service events                          |

### Execution Receipt Meta

ExecutionReceiptMeta contains the fields from the Execution Receipts that vary from one executor to another

> (assuming they commit to the same result).

Reviewer: this is more likely to cause confusion in the docs. let's leave it out
```suggestion
```
